File: 1614.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
    let ans = 0, maxAns = 0;
    for (i in s){
        if (s[i] === "(") maxAns= Math.max(++ans,maxAns);
        if (s[i] === ")") ans--;
    }
    return maxAns
};

File: 1614.py
class Solution:
    def maxDepth(self, s: str) -> int:
        ans=0
        maxAns = 0
        for i in range(len(s)):
            if s[i] == "(": 
                ans+=1
                if maxAns < ans: maxAns = ans
            if s[i] == ")": 
                ans-=1
        return maxAns